tesseract-server/src/handlers/flush.rs

```diff
@@ -1,7 +1,61 @@
 use serde_derive::{Serialize, Deserialize};
 
+use lazy_static::lazy_static;
+use log::*;
+use serde_qs as qs;
+
+use actix_web::{
+    HttpRequest,
+    HttpResponse,
+    Result as ActixResult,
+};
+
+use crate::app::AppState;
+use crate::schema_config;
+
 #[derive(Debug, Deserialize, Serialize)]
 pub struct FlushQueryOpt {
     pub secret: String,
 }
 
+pub fn flush_handler(req: HttpRequest<AppState>) -> ActixResult<HttpResponse> {
+    let query = req.query_string();
+    lazy_static!{
+        static ref QS_NON_STRICT: qs::Config = qs::Config::new(5, false);
+    }
+    let query_res = QS_NON_STRICT.deserialize_str::<FlushQueryOpt>(&query);
+    let query = match query_res {
+        Ok(q) => q,
+        Err(err) => {
+            return Ok(HttpResponse::BadRequest().json(err.to_string()));
+        },
+    };
+
+    let db_secret = match &req.state().env_vars.flush_secret {
+        Some(db_secret) => db_secret,
+        None => { return Ok(HttpResponse::Unauthorized().finish()); }
+    };
+
+    if query.secret == *db_secret {
+        info!("Flush internal state");
+
+        // Read schema again
+        let schema = match schema_config::read_schema() {
```

Comment:
Same story here for reading the schema; I think that we can use the schema filepath which is now passed into state. For me, it's a bit more explicit, plus I think it will avoid reading the env var both outside and inside the `read_schema` fn (from what I see in `main`)